src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViews.component.ts

```diff
@@ -0,0 +1,283 @@
+	@Input() model: NotebookModel;
+	@Input() activeView: INotebookView;
+	@Input() views: NotebookViewsExtension;
+
+	@ViewChild('container', { read: ElementRef }) private container: ElementRef;
+	@ViewChild('viewsToolbar', { read: ElementRef }) private viewsToolbar: ElementRef;
+	@ViewChild(NotebookViewsGridComponent) private gridstack: NotebookViewsGridComponent;
+	@ViewChildren(CodeCellComponent) private codeCells: QueryList<CodeCellComponent>;
+	@ViewChildren(TextCellComponent) private textCells: QueryList<TextCellComponent>;
+
+	protected _actionBar: Taskbar;
+	public previewFeaturesEnabled: boolean = false;
+	private _modelReadyDeferred = new Deferred<NotebookModel>();
+
+	private _scrollTop: number;
+
+	constructor(
+		@Inject(IBootstrapParams) private _notebookParams: INotebookParams,
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _changeRef: ChangeDetectorRef,
+		@Inject(IKeybindingService) private keybindingService: IKeybindingService,
+		@Inject(IContextMenuService) private contextMenuService: IContextMenuService,
+		@Inject(INotificationService) private notificationService: INotificationService,
+		@Inject(INotebookService) private notebookService: INotebookService,
+		@Inject(IConnectionManagementService) private connectionManagementService: IConnectionManagementService,
+		@Inject(IConfigurationService) private _configurationService: IConfigurationService,
+		@Inject(IEditorService) private _editorService: IEditorService
+	) {
+		super();
+		this._register(this._configurationService.onDidChangeConfiguration(e => {
+			this.previewFeaturesEnabled = this._configurationService.getValue('workbench.enablePreviewFeatures');
+		}));
+	}
+
+	public get notebookParams(): INotebookParams {
+		return this._notebookParams;
+	}
+
+	public get id(): string {
+		return this.notebookParams.notebookUri.toString();
+	}
+
+	isDirty(): boolean {
+		return this.notebookParams.input.isDirty();
+	}
+	isActive(): boolean {
+		return this._editorService.activeEditor ? this._editorService.activeEditor.matches(this.notebookParams.input) : false;
+	}
+	isVisible(): boolean {
+		let notebookEditor = this.notebookParams.input;
+		return this._editorService.visibleEditors.some(e => e.matches(notebookEditor));
+	}
+	executeEdits(edits: ISingleNotebookEditOperation[]): boolean {
+		throw new Error('Method not implemented.');
+	}
+	async runCell(cell: ICellModel): Promise<boolean> {
+		await this.modelReady;
+		let uriString = cell.cellUri.toString();
+		if (this.model.cells.findIndex(c => c.cellUri.toString() === uriString) > -1) {
+			this.selectCell(cell);
+			return cell.runCell(this.notificationService, this.connectionManagementService);
+		} else {
+			return Promise.reject(new Error(localize('cellNotFound', "cell with URI {0} was not found in this model", uriString)));
+		}
+	}
+
+	public async runAllCells(startCell?: ICellModel, endCell?: ICellModel): Promise<boolean> {
```

Comment:
It seems like there's a lot of duplicated code here. Why not make an abstract notebookEditor class with the common code that both this and the normal notebook editor inherit from?

Reply:
Ok. I can look into it. I have another PR that I am working on for the NB Views UI. I can address it there if that's ok with you?

Reply:
I'd rather that work be done in its own PR TBH, but whatever works for you.